File: server/common/log/logger.go
package log

// ------------------------------
// FUNCIONES
// ------------------------------
import (
	"log"
	"os"
	"time"
)

// ------------------------------
// PARÁMETROS Y VARIABLES
// ------------------------------

// Tipo para el nivel de los logs.
type LogLevel string

// Niveles de severidad de los logs.
const (
	INFO  LogLevel = "INFO"
	DEBUG LogLevel = "DEBUG"
	ERROR LogLevel = "ERROR"
	WARN  LogLevel = "WARN"
)

// ------------------------------
// FUNCIONES
// ------------------------------

// Crea un logger personalizado.
func CreateLogger(name string, level LogLevel) *log.Logger {
	// Crea el prefijo.
	prefix := time.Now().Format("2006/01/02 15:04:05") + " - " + string(level) + " - [" + name + "]: "
	// Crea y retorna el logger.
	return log.New(os.Stdout, prefix, 0)
}

File: product/services/rag/src/common/logger/logger.py
# ------------------------------------------------------------
# Autor: Pablo González García
# Fecha: 23 Septiembre 2025
# Descripción: Contiene las funcionalidades principales
#       del módulo interno de logger.
# ------------------------------------------------------------


# ------------------------------
# IMPORTS
# ------------------------------

# Estándar:
import logging
from logging.handlers import RotatingFileHandler
from typing import Optional

# Internas:
from .model import LoggerConfig


# ------------------------------
# FUNCIONES
# ------------------------------

def create_logger(name:str, logger_config:Optional[LoggerConfig] = None) -> logging.Logger:
    """
    Crea y retorna un logger.

    Args:
        name (str):                             Nombre del logger a crear.
        logger_config (Optional[LoggerConfig]): Configuración del logger. Por defecto
            es None y se inicializa con los valores por defecto.
    """
    # Comprueba si logger_config es nulo.
    if not logger_config:
        # Crea una configuración por defecto.
        logger_config = LoggerConfig()

    # Crea el logger a retornar.
    logger:logging.Logger = logging.getLogger(name=name)
    # Establece el nivel mínimo de severidad del logger.
    logger.setLevel(logger_config.level)
    
    # Crea el formatter para los logs.
    formatter:logging.Formatter = logging.Formatter(logger_config.format,
                                                    datefmt=logger_config.datefmt)

    # Crea el handler de consola.
    console_handler = logging.StreamHandler()
    console_handler.setLevel(logger_config.level)
    console_handler.setFormatter(formatter)
    # Añade el handler.
    logger.addHandler(console_handler)

    # Comprueba si file_path no es nulo.
    if logger_config.file_path:
        # Crea el handler de consola.
        file_handler:RotatingFileHandler = RotatingFileHandler(filename=logger_config.file_path,
                                                               maxBytes=logger_config.max_bytes,
                                                               backupCount=logger_config.backup_count)
        file_handler.setLevel(logger_config.level)
        file_handler.setFormatter(formatter)
        # Añade el handler.
        logger.addHandler(file_handler)

    # Retorna el logger.
    return logger